### ISSUE TITLE
Private build

### ISSUE THREAD

**Issue by gabrielmd** [jupyterlab/jupyterlab#4534]
Hi guys,

I am trying to make some changes in the Jupyter Lab core, and I want to share my changes with my co-workers. In short, I would like to have an internal version with some customizations that would help our daily job. For example, I wanted to translate it into our language so all the team would be able to understand better.

My question is: how could I make a new build? 

I only see that changes locally if I run "jupyter lab --dev-mode," but it wouldn't be enough if I wanted to keep the changes in our private repository. What should I do so that the others will be able to get the code from our repository, run "jupyter lab" and see the updated code?

**Comment by timkpaine** [comment]
@blink1073 and I were discussing this recently. It's hard to make changes to core packages AND install extensions. It would be nice if Core packages were as easily overideable as -extension packages

**Comment by saulshanabrook** [comment]
Yeah it would be nice if they were all treated the same. The builtins could be hidden maybe by default with `labextension list`, but could be uninstall just the same as others.

**Comment by SimonBiggs** [comment]
Maybe instead of hiding them on labextension list, they could be displayed first under a heading "core extensions" followed by the remaining extensions displayed under "custom extensions". By updating or changing a core extension it would move from the core list to the custom list...

**Comment by jasongrout** [comment]
This would also be less confusing if we tracked and displayed the dependencies between extensions. Right now many core extensions depend on many others, and it's difficult to see at a glance what can be disabled and how that affects other extensions.

This also happens with third-party extensions, like ipywidgets, but it's much more of an issue right now with core extensions.

**Comment by jasongrout** [comment]
CC also @vidartf, who has worked in related areas, for example #4410.

**Comment by vidartf** [comment]
It *should* be possible to uninstall core extensions with `jupyter labextension uninstall @jupyterlab/<...>`. As in, there is code there to deal with it. Whether or not it actually works, I cannot attest to.
